.github/workflows/fleet-and-orbit.yml

```diff
@@ -383,6 +383,38 @@ jobs:
       run: |
         "C:\Program Files\Orbit\bin\orbit\orbit.exe" shell -- --json "select * from osquery_info;" | jq -e "if (.[0]) then true else false end"
 
+    - name: Fleet Service Tests
+      shell: powershell
+      run: |
+          #Tests setup
+          $serviceName = "Fleet osquery"
+          $defaultWaitTime = 2
+
+          #Test 1 - Check that the service starts without issues
+          Stop-Service -Name $serviceName
+          Start-Service -Name $serviceName
+          Get-Service -Name $serviceName | %{ if ($_.Status -ne "Running") { throw "Test #1 failed" } }
+
+          #Test 2 - Check that the service stops without issues
+          Stop-Service -Name $serviceName
+          Get-Service -Name $serviceName | %{ if ($_.Status -ne "Stopped") { throw "Test #2 failed" } }
+
+          #Test 3 - Check that no orbit.exe is running after service stop
+          Start-Service -Name $serviceName
+          Stop-Service -Name $serviceName
+          Start-Sleep -Seconds $defaultWaitTime # shutdown wait time
+          Get-Process | %{ if ($_.Name -eq "orbit") { throw "Test #3 failed" } }
+
+          #Test 4 - Check that service starts in less than 3 secs
+          Start-Job { Start-Service -Name $args[0] } -ArgumentList $serviceName #async operation
+          Start-Sleep -Seconds $defaultWaitTime
+          Get-Service -Name $serviceName | %{ if ($_.Status -ne "Running") { throw "Test #4 failed" } }
+
+          #Test 5 - Check that service stops in less than 3 secs
+          Start-Job { Stop-Service -Name $args[0] } -ArgumentList $serviceName #async operation
+          Start-Sleep -Seconds $defaultWaitTime
+          Get-Service -Name $serviceName | %{ if ($_.Status -ne "Stopped") { throw "Test #5 failed" } }       
+
```

Comment:
Really cool tests, hope I learn PowerShell some day :)

Reply:
thanks! Powershell is really useful in a windows environment. It gives a lot of helpers and access to the entire win32 API